src/Components/Browser.JS/src/Rendering/Renderer.ts

```diff
@@ -1,13 +1,18 @@
+/* eslint-disable @typescript-eslint/camelcase */
 import { System_Object, System_String, System_Array, MethodHandle, Pointer } from '../Platform/Platform';
 import { platform } from '../Environment';
 import { RenderBatch } from './RenderBatch/RenderBatch';
-import { BrowserRenderer } from './BrowserRenderer';
+import { BrowserRenderer, StartEndPair } from './BrowserRenderer';
 
-type BrowserRendererRegistry = { [browserRendererId: number]: BrowserRenderer };
+interface BrowserRendererRegistry {
+  [browserRendererId: number]: BrowserRenderer;
+}
 const browserRenderers: BrowserRendererRegistry = {};
 
-export function attachRootComponentToElement(browserRendererId: number, elementSelector: string, componentId: number) {
-  const element = document.querySelector(elementSelector);
+export function attachRootComponentToElement(browserRendererId: number, elementSelector: string | StartEndPair, componentId: number): void {
+
+  const { start, end } = elementSelector as StartEndPair;
```

Comment:
Instead of introducing the new concept of a `StartEndPair`, would it be possible to construct a `LogicalElement` from the nodes we have here, and pass that through instead? I'm trying to understand if there's anything different that `StartEndPair` does for us, but can't spot anything.

Reply:
The selector is a string, the start end pair delimits the nodes we use for removing the initial prerendered content. 

I don't see a way of doing this that doesn't involve changing the entry point for the renderer a bit or repeating work.
1) We could pass in the comment and the selector, then:
  1.1) Resolve the node from some sort of store.
  1.2) Re-scan the document for the start and end pairs.
2) We could simply pass in the origin node, and:
  2.2) Find the end node among its siblings.
  2.3) Access a property on the start node to find the related end node.

I didn't like the idea of repeating work. That is why I passed in the start and end nodes. This affects  a very small subset of the renderer API (the entry point) so I don't think it's a big deal. A better name than start and end pair might be ComponentDelimeters or ComponentDescriptor.

I'm not very worried about this as it doesn't affect functionality or impacts the renderer substantially. If you are ok with it, we can make a decision on how we want things to look, file a bug and I can clean it up later this week/next week. Whatever you say goes here as you are the expert in this part of the codebase.